src/software/simulation/simulator_ball_singleton.cpp

```diff
@@ -39,20 +42,60 @@ float SimulatorBallSingleton::checkValidAndReturnFloat(
 
 float SimulatorBallSingleton::getBallPositionX()
 {
-    return checkValidAndReturnFloat([](auto ball) { return ball->position().x(); });
+    return checkValidAndReturnFloat([](auto ball) {
+        switch (field_side_)
+        {
+            case FieldSide::NEG_X:
+                return ball->position().x();
+            case FieldSide::POS_X:
+                return -ball->position().x();
+            default:
+                throw std::invalid_argument("Unhandled value of FieldSide");
+        }
+    });
```

Comment:
This same chunk of code is used in 8 places, is there anyway we can turn this into a small helper function?

Reply:
Maybe, but not sure that would increase clarity. These blocks are slightly different since they return x position, y position, x velocity, etc. so the values are slightly different.

For example we could pass in a second lambda with just the `ball->velocity().y()` part for example, but I'm not sure that would make the code more readable.

Eg.
```
return checkValidAndReturnFloat([](auto ball) {
    return getInvertedValue(ball->position().x());
}
```

Reply:
fixed